### DISCUSSION TITLE
CropRegion - is it used in Mantis?

### DISCUSSION THREAD

**Discussion by rickshane** [guoyingtao/Mantis#412] [General]
I have been using CropRegion in my test harness to perform relative crops of images at different zoomScales. This is useful when an image has been scaled up or down before applying the crop. However, when I fully rotate an image to 45 degrees, I cannot seem to properly calculate the crop width and height based on the CropRegion data. 

I don't see CropRegion being used in Mantis. I was hoping to compare my results with the Mantis test harness. Did this logic get deprecated at some point? It is a very useful data structure.

I use the following logic to calculate the crop width and height of a rotated image using cropRegion:

```
cropRect.size.width = sqrt(pow(cropRegion.topRight.x-cropRegion.topLeft.x, 2) + pow(cropRegion.topLeft.y-cropRegion.topRight.y, 2)) * scaledImageSize.width
                
cropRect.size.height = sqrt(pow(cropRegion.bottomLeft.x-cropRegion.topLeft.x, 2) + pow(cropRegion.bottomLeft.y-cropRegion.topLeft.y, 2)) * scaledImageSize.height
```

Or is there a better mechanism to crop an image that has first been scaled?

Thanks,

Rick

**Comment by guoyingtao** [comment]
@rickshane 
The items in `CropRegion` are normalized values which are between 0 and 1 and `CropRegion` is part of `CropInfo` like below

```Swift
public typealias CropInfo = (
    translation: CGPoint,
    rotation: CGFloat,
    scaleX: CGFloat,
    scaleY: CGFloat,
    cropSize: CGSize,
    imageViewSize: CGSize,
    cropRegion: CropRegion
)
```

Just want to know if the cropRect size you mentioned is the cropped image size? If so, you can use `CropViewController.getExpectedCropImageSize()`

**Comment by rickshane** [comment]
I solved it. I was not multiplying the sides by the scaled image width and height before squaring and calculating the hypotenuse. The correct equations are:

```
cropRect.size.width = sqrt(pow((cropRegion.topRight.x-cropRegion.topLeft.x) * scaledImageSize.width, 2) + pow((cropRegion.topLeft.y-cropRegion.topRight.y) * scaledImageSize.height, 2))
                    
cropRect.size.height = sqrt(pow((cropRegion.bottomLeft.x-cropRegion.topLeft.x) * scaledImageSize.width, 2) + pow((cropRegion.bottomLeft.y-cropRegion.topLeft.y) * scaledImageSize.height, 2))
```
